### ISSUE TITLE
Request to review acc6.txt

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/ACC#17]
I am done doing addition of features to acc.txt for now on dev server.
Time to review the progress so far.

I request @funderburkjim to review the current acc.txt (equivalent to acc6.txt) and see if we are missing some information or not.

Explanation is attached below.

# acc1.txt and acc2.txt
Generated from acc_orig_utf8_slp1.txt.
See ../pywork/update.sh.

# acc3.txt
Generated from acc2.txt by adding meta-line details.
See ../pywork/correctionwork/issue-cologne-130/redo.sh.

# acc4.txt
Generated from acc3.txt by adding subject and person tags.
See ../pywork/correctionwork/issue-cologne-142/subject_tag_addition.py.

# acc5.txt
Generated from acc4.txt by adding literary source tags (ls).
See ../pywork/correctionwork/issue-cologne-148/catalogue_tag_addition.py.

# acc6.txt
Generated from acc5.txt by adding internal references. 
See ../pywork/correctionwork/issue-acc-2/hw_int_ref.py.

# acc.txt
The most feature-rich item goes here. Currently acc6.txt is equal to acc.txt.
In future, there may be acc7, acc8, and so on.
The highest number should be carried to acc.txt.

**Comment by drdhaval2785** [comment]
If acc.txt is found OK, it may be a taken as base file in live server.

**Comment by funderburkjim** [comment]
I'll review in a few days.  You've done a lot of markup, and I want to understand.

Here are some preliminary questions:
* What is the objective of the markup ?  (a verbal summary of your goals in adding the markup)
* I think you've done the markup in the accX.txt files.
   * Have you remade acc.xml  (i.e., does make_xml.py work properly with the added markup)?
* Have you adjusted the display (web/webtc/disp.php) to properly handle the extra xml markup?

**Comment by gasyoun** [comment]
> Have you adjusted the display (web/webtc/disp.php) to properly handle the extra xml markup?

Guess not yet.

**Comment by drdhaval2785** [comment]
> What is the objective of the markup ?

1. Subject tag addition
1.1. `<ab type="subj">` - Find out subjects such as lex, gr, vaid, kāvya etc.
1.2. `<ab type="pers">` - Find out relations like son, father, disciple etc
2. Catalogue tag addition
Find out the literary sources of a particular work e.g. Stein, Kāvyamālā, Oudh etc. 
3. Internal references
A dictionary entry may refer to some known headword e.g. AnandarAya entry has `son of Nṛsiṃharāya` which we want to capture like `<ab type="pers">son</ab> of <ab type="hw" value="nfsiMharAya">Nṛsiṃharāya</ab>` because it refers to the headword nfsiMhAcArya.

> Have you remade acc.xml?

No

> Have you adjusted the display (web/webtc/disp.php) to properly handle the extra xml markup?

No.

I wanted you to see the acc6.txt file and give your input whether it is OK to add markup like this or not. If yes, then downstream modifications can be made. If not, there is no point doing these changes and redo them again.

**Comment by funderburkjim** [comment]
>  whether it is OK to add markup like this 

As a general principle, it is OK.  

From the examples, I conclude that you are adding only markup.
i.e., if we removed all the `<ab...>,</ab>,<ls>,</ls>` tags from acc6.txt, then we would get exactly acc3.txt.  Right?

Adding only markup is good.

**Comment by funderburkjim** [comment]
### acc.xml and acc.dtd needed.

We can't fully evaluate the xml markup additions of acc6.txt (= acc.txt currently), until we
* generate acc.xml (using make_xml.py), 
* modify acc.dtd 
* validate acc.xml against acc.dtd using xmllint (or, on Windows, xmlvalidate.py, which uses lxml module.)
  * xmlvalidate.py is in Cologne repository [here](https://github.com/sanskrit-lexicon/Cologne/tree/master/enhancements/code)
  * On my local PC, here is the path to xmlvalidate.py (using git-bash syntax for path)
    ``` /c/xampp/htdoc/cologne/ ```
  * If you use xmlvalidate.py, suggest you put it here, in particular do not put it into acc repository.


@drdhaval2785   You need to do this step next.

**Comment by gasyoun** [comment]
**Question type:** editorial-policy / status review

**Evidence found:**
- The current [acc.txt](https://github.com/sanskrit-lexicon/csl-orig/blob/master/v02/acc/acc.txt) in csl-orig has **214,397 lines** and contains **zero** `<ab>` tags and **zero** `<ls>` tags.
- This means the markup additions described for acc3–acc6 (subject tags `<ab type="subj">`, person tags `<ab type="pers">`, literary-source `<ls>` tags, and internal reference `<ab type="hw">` tags) are **not present** in the current canonical file.
- Only two commits touch acc.txt in csl-orig: one correction commit and one dangling-`<ls>` merge fix — neither is the acc6 markup merge.
- The versioned files acc1.txt through acc6.txt are not present in csl-orig v02/acc/; only acc.txt and auxiliary files (acc_front.txt, acc_back.txt, acc_middle.txt, acc-meta2.txt, acc_hwextra.txt) exist.
- Related open issue [#12](https://github.com/sanskrit-lexicon/ACC/issues/12) (downstream modifications for XML and display) is also open, confirming the acc6 pipeline was never completed downstream.

**Assessment:**
Appears stalled / partially resolved: The conversation in 2017 ended with funderburkjim approving the markup approach in principle but requesting XML generation + DTD update + xmllint validation before the changes could be accepted. That step was never reported as completed. The acc6 markup was consequently never merged into the canonical acc.txt in csl-orig. The issue is effectively blocked on completing the XML validation step from @funderburkjim's [last comment](https://github.com/sanskrit-lexicon/ACC/issues/17#issuecomment-327297888).

**Suggested next step:** Locate the acc6.txt file (likely in the ACC pywork correction directory or a dev server backup), run make_xml.py to generate acc.xml, update acc.dtd to cover the new `<ab type="subj|pers|hw">` elements, validate with xmllint or xmlvalidate.py, and — if clean — copy acc6.txt to csl-orig/v02/acc/acc.txt and commit. Issue [#12](https://github.com/sanskrit-lexicon/ACC/issues/12) (display/PHP handling of the new tags) can be addressed in parallel.
